api/api-pipeline-config-v6/src/main/java/com/thoughtworks/go/apiv6/admin/pipelineconfig/representers/PipelineConfigRepresenter.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2018 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.apiv6.admin.pipelineconfig.representers;
+
+import com.thoughtworks.go.api.base.OutputListWriter;
+import com.thoughtworks.go.api.base.OutputWriter;
+import com.thoughtworks.go.api.representers.ErrorGetter;
+import com.thoughtworks.go.api.representers.JsonReader;
+import com.thoughtworks.go.apiv6.admin.pipelineconfig.representers.materials.MaterialRepresenter;
+import com.thoughtworks.go.apiv6.admin.pipelineconfig.representers.trackingtool.TrackingToolRepresenter;
+import com.thoughtworks.go.apiv6.shared.representers.EnvironmentVariableRepresenter;
+import com.thoughtworks.go.apiv6.shared.representers.configorigin.ConfigRepoOriginRepresenter;
+import com.thoughtworks.go.apiv6.shared.representers.configorigin.ConfigXmlOriginRepresenter;
+import com.thoughtworks.go.apiv6.shared.representers.stages.StageRepresenter;
+import com.thoughtworks.go.config.*;
+import com.thoughtworks.go.config.exceptions.UnprocessableEntityException;
+import com.thoughtworks.go.config.materials.MaterialConfigs;
+import com.thoughtworks.go.config.remote.ConfigOrigin;
+import com.thoughtworks.go.config.remote.FileConfigOrigin;
+import com.thoughtworks.go.config.remote.RepoConfigOrigin;
+import com.thoughtworks.go.spark.Routes;
+import org.bouncycastle.crypto.InvalidCipherTextException;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class PipelineConfigRepresenter {
```

Comment:
Perhaps break this down into pipeline/stage/job/environmentVariable/params representers?

This feels messy from the looks of it.

Reply:
It is broken down to stage, job, env, param representer? Can you please explain what you mean?